adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala

```diff
@@ -3046,4 +3188,41 @@ class ADAMContext(@transient val sc: SparkContext) extends Serializable with Log
       loadParquetFragments(pathName, optPredicate = optPredicate, optProjection = optProjection)
     }
   }
+
+  /**
+   * Test if Parquet files are partitioned
+   *
+   * @param filePath Path in which to look for partitioned flag.
+   * @return Return True if partitioned flag found, False otherwise.
+   */
+
+  def checkPartitionedParquetFlag(filePath: String): Boolean = {
+    val path = new Path(filePath, "_isPartitionedByStartPos")
+    val fs = path.getFileSystem(sc.hadoopConfiguration)
+    fs.exists(path)
+  }
+
+  /**
+   * Returns a query string used to filter a dataset based on an input list of ReferenceRegion
+   *
+   * @param regions list of regions to include in query
+   * @param partitionSize size of genomic range partitions used when writing parquet
+   * @param addChrPrefix add the prefix "chr" to the contig names - use if the original loaded data and sequence
+   *                     dictionary used  "chr" prefix
+   * @return
+   */
+
+  def referenceRegionsToDatasetQueryString(regions: Iterable[ReferenceRegion], partitionSize: Int = 1000000): String = {
+
+    var regionQueryString = "(contigName=" + "\'" + regions.head.referenceName + "\' and posBin >= \'" +
```

Comment:
This will throw if `regions.isEmpty`, suggest:

```
regions.map(r => {
  // logic for a single reference region goes here
}).mkString(" or " )
```